### PR TITLE
Add C++ Solution + Editorial to Teamwork (Gold)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
the difficulty inflation...

---
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-863.mdx [hunk 1]
```diff
@@ -2,17 +2,60 @@
...
+		// by not joining it to a team, the default value is just skill[i]
+		int cur = skill[i];
 
+		for (int j = i; j >= (i - k + 1) && ~j; j--) {
+			cur = max(cur, skill[j]);
+			/*
+			 * update answer: dp[i] = max(dp[i], 
+			 * the skill of joining this cow to a team + all of the previous sums)
+			 */
+			if (j > 0) {
```
Comment (SansPapyrus683) [inline solutions/usaco-863.mdx:48]: if this is supposed to check for just -1, be more explicit
```suggestion
			if (j > -1) {
```

File: solutions/usaco-863.mdx [hunk 2]
```diff
@@ -2,17 +2,60 @@
...
+				dp[i] = max(dp[i], dp[j - 1] + (cur * (i - j + 1)));	
+			} else {
+				dp[i] = max(dp[i], cur * (i - j + 1));
+			}
+		}
+	}
+	cout << dp[n - 1] << '\n';
```
Comment (SansPapyrus683) [inline solutions/usaco-863.mdx:55]: (optional)
```suggestion
	cout << dp[n - 1] << endl;
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-863.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/usaco-863.mdx:45]: it's checking whether it's greater than 0...

if it's equal to zero, then we don't subtract one, or else it'll go out of bounds.

Comment (SansPapyrus683) [reply solutions/usaco-863.mdx:45]: ok cool

**Review by SansPapyrus683 — APPROVED** [review]

**Review by TruaShamu — APPROVED** [review]

**Review by jessechoe10 — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
ryan's pr's are god's gifts upon this world

**Pull request merged**